src/domain/membership/information/MembershipInformation.tsx

```diff
@@ -3,75 +3,316 @@ import { useTranslation } from 'react-i18next';
 import { QRCode } from 'react-qrcode-logo';
 import { Button } from 'hds-react';
 
-import { MembershipInformation as MembershipInformationTypes } from '../../../graphql/generatedTypes';
+import {
+  MembershipInformation_myYouthProfile,
+  MembershipInformation_myYouthProfile_profile,
+  MembershipStatus,
+} from '../../../graphql/generatedTypes';
 import LinkButton from '../../../common/components/linkButton/LinkButton';
-import Text from '../../../common/components/text/Text';
 import convertDateToLocale from '../../../common/helpers/convertDateToLocale';
-import getFullName from '../helpers/getFullName';
-import styles from './membershipInformation.module.css';
+import MembershipPageLayout from '../MembershipPageLayout';
+import Membership from '../MembershipUtility';
 
-interface Props {
+type MembershipExpiryTitleProps = {
+  membershipExpirationDate: string;
+  memberShipStatus: MembershipStatus;
+  membershipRenewable: boolean;
+};
+
+function MembershipExpiryTitle({
+  membershipExpirationDate,
+  memberShipStatus,
+  membershipRenewable,
+}: MembershipExpiryTitleProps) {
+  const { t } = useTranslation();
+
+  const isRenewable = membershipRenewable;
+  const isExpired = Membership.getIsExpired(memberShipStatus);
+  const isActive = Membership.getIsActive(memberShipStatus);
+  const isRenewing = Membership.getIsRenewing(memberShipStatus);
+  const isPending = Membership.getIsPending(memberShipStatus);
```

Comment:
These variables are declared four times in this file. I'd consider combining them under one getter function e.g `Membership.getStatus` that would return an object containing all the values. This could then be destructured to get the variables `isExpired, isActive` etc.

Reply:
I agree that the API would be nicer for this usecase in that there wouldn't be as many lines. However, I'm hesitant to create the extra level of abstraction, because I am not sure about how well it would fare with other usecases. At this point I feel like the change would be mostly syntax sugar as there would still be as many declarations, but with the chance that it'd be inconvenient in some other context later on.

However, I'm willing to make this change if there's some perspective I'm not noticing.

Reply:
True, amount of declarations would be the same, but imo would clean up syntax a little bit and declaring them one by one feels kind of "heavy". We can leave it as it is for now, but it might be worth it the future if we implement other features that rely on multiple statuses.